iam.tf

```
@@ -14,6 +14,18 @@ module "external_secrets_irsa" {
   }
 }
 
-data "aws_secretsmanager_secret" "batcave-private-registry" {
-  name = "batcave/registry-artifactory.cloud.cms.gov"
+locals {
+    secret_arns = [ for name in local.secret_names : "arn:aws:secretsmanager:${var.aws_region}:${var.aws_id}:secret:${name}"
+    ]
+    secret_names = [
+        "private-registry",
+        "batcave/argocd-config-batcave-dev",
+        "batcave/grafana-secret-batcave-dev",
+        "batcave/sso-secret-batcave-dev",
+        "batcave/istio-secret-batcave-dev",
+        "batcave/sonar-registry-credentials-batcave-dev",
+        "batcave/sonar-agent-api-key-batcave-dev",
+        "batcave/kiali-batcave-dev",
+        "batcave/gitlab-rails-secret-s3-batcave-dev"
```

Remove "batcave-dev" suffix
```suggestion
        "batcave/argocd-config",
        "batcave/grafana-secret",
        "batcave/sso-secret",
        "batcave/istio-secret",
        "batcave/sonar-registry-credentials",
        "batcave/sonar-agent-api-key",
        "batcave/kiali",
        "batcave/gitlab-rails-secret-s3"
```

---

So the way i have the migration script now is it checks the aws profile and if its batcave-dev, it adds batcave-dev to the end of the secret name. If not it adds the cluster name to the end. Are you saying to just remove the suffix for the dev env?

---

yeah, so that way these names can be consistent across environments.